### PR TITLE
Feature set subscription message

### PR DESCRIPTION
**Pull request by Frank-Wiebeler**
Added SubScriptionMessageService as this was the last service not available

### CODE REVIEW

**Comment by saschadoemer** [comment]
I see, you could not see the comments, because the review was not submitted. Therefore the comments from a month ago did not show up.

File: agrirouter-api-java-api/src/main/java/com/dke/data/agrirouter/api/service/messaging/SetSubscriptionService.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.dke.data.agrirouter.api.service.messaging;
+
+import com.dke.data.agrirouter.api.service.parameters.SetSubscriptionParameters;
+
+public interface SetSubscriptionService extends MessagingService<SetSubscriptionParameters> {}
```
[the remark this answers is not in the document]
Comment (Frank-Wiebeler) [reply agrirouter-api-java-api/src/main/java/com/dke/data/agrirouter/api/service/messaging/SetSubscriptionService.java]: I was wondering about SetCapabilities at first, but now that we have this "error", I think it makes sense to call it SetSubscriptionService as well. These 2 are the only commands actually setting parameters of the endpoint, therefore Set would make sense. 

Would vote for leaving this as it is, as this naming is something, I can describe the reason for. If we call it SubscriptionService, I could describe "SetCapabilitiesService" only as an error. Don't think that this is an advantage ;-)

File: agrirouter-api-java-api/src/main/kotlin/com/dke/data/agrirouter/api/service/parameters/SetSubscriptionParameters.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Parameters class. Encapsulation for the services.
+ */
+@ToString
+class SetSubscriptionParameters : AbstractParameterBase(), ParameterValidation {
+
+    @NotNull
+    lateinit var onboardingResponse: OnboardingResponse
+
+
```
[the remark this answers is not in the document]
Comment (Frank-Wiebeler) [reply agrirouter-api-java-api/src/main/kotlin/com/dke/data/agrirouter/api/service/parameters/SetSubscriptionParameters.kt]: Done

File: agrirouter-api-java-api/src/main/kotlin/com/dke/data/agrirouter/api/service/parameters/SetSubscriptionParameters.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    class Subscription : ParameterValidation {
+        @NotNull
+        lateinit var technicalMessageType: TechnicalMessageType
+
```
[the remark this answers is not in the document]
Comment (Frank-Wiebeler) [reply agrirouter-api-java-api/src/main/kotlin/com/dke/data/agrirouter/api/service/parameters/SetSubscriptionParameters.kt]: Done

File: agrirouter-api-java-api/src/main/kotlin/com/dke/data/agrirouter/api/factories/impl/SubscriptionMessageContentFactory.kt [hunk 4]
```diff
@@ -10,14 +10,15 @@ import java.util.*
...
 class SubscriptionMessageContentFactory {
 
-    fun message(vararg parameters: SubscriptionMessageParameters): ByteString {
-        parameters.forEach { p -> p.validate() }
+    fun message(parameters: SubscriptionMessageParameters): ByteString {
+        parameters.validate()
         val messageContent = SubscriptionOuterClass.Subscription.newBuilder()
-        Arrays.stream(parameters).forEach { p ->
+
+        for ( entry in parameters.list){
```
[the remark this answers is not in the document]
Comment (Frank-Wiebeler) [reply agrirouter-api-java-api/src/main/kotlin/com/dke/data/agrirouter/api/factories/impl/SubscriptionMessageContentFactory.kt]: Was done for some debugging tests and forgot to revert; Now done with kotlins .forEach as done in the Capabilities File.

File: agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/messaging/rest/SetSubscriptionServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    sendMessageParameters.setEncodedMessages(
+        Collections.singletonList(encodeMessageResponse.getEncodedMessage()));
+
+    MessageSenderResponse response = this.sendMessage(sendMessageParameters);
+
+    this.assertStatusCodeIsOk(response.getNativeResponse().getStatus());
+    return encodeMessageResponse.getApplicationMessageID();
+  }
+
+  private EncodeMessageResponse encodeMessage(SetSubscriptionParameters parameters) {
```
[the remark this answers is not in the document]
Comment (Frank-Wiebeler) [reply agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/messaging/rest/SetSubscriptionServiceImpl.java]: Let's wait for merge of MQTT and than I'll adjust this one the same way all the other Services work.

File: agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/onboard/cloud/OnboardingServiceImpl.java [hunk 6]
```diff
@@ -1,5 +1,8 @@
 package com.dke.data.agrirouter.impl.onboard.cloud;
 
+import static com.dke.data.agrirouter.impl.messaging.rest.MessageFetcher.DEFAULT_INTERVAL;
```
[the remark this answers is not in the document]
Comment (Frank-Wiebeler) [reply agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/onboard/cloud/OnboardingServiceImpl.java]: Keeping the order as in all other files; the static imports are always the first ones directly after package. By the way: the order is also like this in the MQTT Implementation Pull Request

**Comment by Frank-Wiebeler** [comment]
Closing this one and creating a new one instead for adding the MQTT Functionality and merging with an updated develop branch

**Pull request closed without merging**